### PR TITLE
docs(emqx_machine): Improve README

### PR DESCRIPTION
**Pull request by ieQu1**
### <samp>🤖 Generated by Copilot at 8dcfa9c</samp>

Updated `apps/emqx_machine/README.md` to reflect the correct application name and provide more information about its functionality and usage.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_machine/README.md [hunk 1]
```diff
@@ -1,5 +1,48 @@
-# EMQX Machine
+# emqx_machine
```
Comment (thalesmg) [inline apps/emqx_machine/README.md:1]: Q: should this be the slug for the application?  I had understood that it should be a "pretty" version of the name :thinking:

File: apps/emqx_machine/README.md [hunk 2]
```diff
@@ -1,5 +1,48 @@
...
+
+# Documention links
+
+Configuration: [node.global_gc_interval](https://www.emqx.io/docs/en/v5.0/configuration/configuration-manual.html#node-and-cookie)
+
+# Configurations
+
+The following application environment variables are used:
+
+- `emqx_machine.global_gc_interval`: interval at which global GC is ran
```
Comment (thalesmg) [inline apps/emqx_machine/README.md:42]:
```suggestion
- `emqx_machine.global_gc_interval`: interval at which global GC is run
```

**Review by thalesmg — APPROVED** [review]

File: apps/emqx_machine/README.md (hunk 2 above)
Comment (ieQu1) [reply apps/emqx_machine/README.md]: Good catch!

File: apps/emqx_machine/README.md (hunk 1 above)
Comment (ieQu1) [reply apps/emqx_machine/README.md]: Thanks, fixed.

**Review by thalesmg — APPROVED** [review]

**Pull request merged**